docs/frontend-system/architecture/07-routes.md

```diff
@@ -8,143 +8,420 @@ description: Frontend routes
+```yaml
+# app-config.yaml
+app:
+  routes:
+    bindings:
+      plugin.catalog.externalRoutes.createComponent: plugin.scaffolfer.routes.createComponent
+```
 
-// app
+Or via code, in the file where the app is created:
 
-import {catalogPlugin} from '@backstage/plugin-catalog'
+```tsx
+// packages/app/src/App.tsx
+import { createApp } from '@backstage/frontend-app-api';
+import catalog from '@backstage/plugin-catalog';
+import scaffolder from '@backstage/plugin-scaffolder';
 
 const app = createApp({
-  bindRoutes({bind}) {
-    bind(scaffolderPlugin, {
-      catalogIndex: catalogPlugin.routes.index,
-    })
+  bindRoutes({ bind }) {
+    bind(catalog.externalRoutes, {
+      createComponent: scaffolder.routes.createComponent,
+    });
   },
-})
-*/
+});
+
+export default app.createRoot();
+
+// packages/app/src/index.ts
+import ReactDOM from 'react-dom/client';
+import app from './App';
+
+ReactDOM.createRoot(document.getElementById('root')!).render(app);
+```
+
+Given the above binding, using `useRouteRef(createComponentRouteRef)` within the Catalog plugin will let us create a link to whatever path the Scaffolder create component page is mounted at.
+
+Note that we are not importing and using the RouteRefs directly in the app, and instead rely on the plugin instance to access routes of the plugins. This is a new convention that was introduced to provide better namespacing and discoverability of routes, as well as reduce the number of separate exports from each plugin package. this his indirection in the routing is particularly useful for open source plugins that need to leave flexibility in how they are integrated.
+
+Another thing to note is that this indirection in the routing is particularly useful for open source plugins that need to leave flexibility in how they are integrated. For plugins that you build internally for your own Backstage application, you can choose to go the route of direct imports or even use concrete routes directly. Although there can be some benefits to using the full routing system even in internal plugins. It can help you structure your routes, and as you will see further down it also helps you manage route parameters.parameters.
+
+### Optional External Route References
+
+When creating an ExternalRouteRef it is possible to mark it as optional:
+
+```tsx
+// plugins/catalog/src/routes.ts
+import {
+  createRouteRef,
+  createExternalRouteRef,
+} from '@backstage/frontend-plugin-api';
+
+const rootRouteRef = createRouteRef();
+const createComponentRouteRef = createExternalRouteRef({
+  optional: true,
+});
+```
+
+An external route that is marked as optional is not required to be bound in the app, allowing it to be used as a switch for whether a particular link should be displayed or action should be taken.
+
+When calling useRouteRef with an optional external route, its return signature is changed to RouteFunc | undefined, allowing for logic like this:
+
+```tsx
```

Comment:
I think this example to be condensed down into just an example for usage within an individual component instead?

Reply:
Done 🙌🏻